### PR TITLE
MM-27212: Cypress/e2e: Add user to channel from popover

### PR DESCRIPTION
**Pull request by agnivade**
This PR automates
- MM-T3
- MM-T4
- MM-T5
- MM-T6
- MM-T7
- MM-T8
- MM-T9
- MM-T1

https://mattermost.atlassian.net/browse/MM-27212

### CODE REVIEW

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (agnivade) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:34]: @saturninoabril - you were right. Clicking it did disable it. Removed it now. Not sure why it wasn't working yesterday.

**Comment by saturninoabril** [comment]
Thanks @agnivade! Could you please check again if you're getting consistent passing results? I've got several failing tests running the spec.
Screen Shot 2020-09-08 at 7 23 26 PM

**Comment by agnivade** [comment]
@saturninoabril - Can you tell me what is failing on T4? I think it's a cascading failure from there. I cannot seem to repro it here.

**Comment by saturninoabril** [comment]
It failed on checking "Manage Public Channels".
Screen Shot 2020-09-08 at 10 30 06 PM

- Do you need license to run each test? If some require license, then I suggest to group and move into a separate spec file with `cy.apiRequireLicense()` check at before hook.
- Each test seemed to be dependent with previous test. I suggest to run the test with `it.only` and see if it'll work as expected. You might also need to reset before each test. Below might help,
```javascript
beforeEach(() => {
    cy.apiAdminLogin();
    cy.apiResetRoles();
});
```

**Comment by agnivade** [comment]
> Each test seemed to be dependent with previous test.

Yep, that needs to be fixed. I'll look into it.

**Comment by agnivade** [comment]
@saturninoabril - should be much better now. PTAL.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (streamer45) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:492]: Using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) (string in backticks) might improve the readability a bit.

**Review by streamer45 — APPROVED** [review]
This was a lot of work. Nicely done :+1: 
Left one small suggestion.

**Comment by saturninoabril** [comment]
@agnivade I'm still seeing consistent failing tests.
Screen Shot 2020-09-09 at 6 30 08 PM

**Comment by agnivade** [comment]
Woops, this is bad. I think somehow my environment gets polluted with other things. I'll dig some more.

**Comment by agnivade** [comment]
Hey @saturninoabril - I tried multiple runs with a fresh DB and it passes every time. Are you getting the same error on T4 as you saw earlier here https://github.com/mattermost/mattermost-webapp/pull/6395#issuecomment-688925963?

**Comment by saturninoabril** [comment]
Hmm, not sure why it turned out like that but here's the latest run on me.
Screen Shot 2020-09-09 at 6 30 08 PM

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (saturninoabril) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:119]: Looks like this one requires another click to fully uncheck the "Manage Public Channels". Initial state has "Convert Channels" uncheck on me and so first click makes the entire "Manage Public Channels" checked.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (saturninoabril) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:84]: I see this repeated in several test blocks. Maybe move to `beforeEach`? Other that don't require it might be better to move in a separate spec file.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:119]: Ah looks like that was it. I had an outdated copy of the webapp. Fixing.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (saturninoabril) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:241]: Same here for another click to fully uncheck.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (saturninoabril) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:408]: And here.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js]: done

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js]: done

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:84]: Hmm .. it might look a bit weird to have T5 and T8 in a separate file and the rest in another, just to de-duplicate 2 lines. I think this is fine as is. And groups all related tests in a single file. 0/5.

Comment (saturninoabril) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:84]: I'll defer to you but just saying for test maintainability where we spend most of the time after the script is merged. Also, on the same note, making each test isolated will help us easily debug the script in the future. You'll know it's isolated if you can run the test successfully after adding `it.only` on one `it` block.

**Review by saturninoabril — APPROVED** [review]
Thanks @agnivade for updating. LGTM, tested and all passed.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js
Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:84]: No worries, I'll move it out. The tests are not 100% isolated because the posts created are shared between them but the user roles being reset and now some more tests being moved out should make them stable.

**Comment by agnivade** [comment]
@saturninoabril - I split them into different spec files now and also added check for the convert channel checkbox. Could you please run the tests once more from your end to confirm everything is okay? Thanks.

**Comment by saturninoabril** [comment]
Thanks @agnivade. Tested and all passed.

**Review by lindy65 — APPROVED** [review]
Thanks @agnivade - looks good to me 👍

**Pull request merged**
